src/Cli.Tests/ValidateConfigTests.cs

```diff
@@ -60,4 +60,20 @@ public void TestConfigWithInvalidConfigProperties()
         bool isConfigValid = ConfigGenerator.IsConfigValid(validateOptions, _runtimeConfigLoader!, _fileSystem!);
         Assert.IsFalse(isConfigValid);
     }
+
+    /// <summary>
+    /// This method validates that the IsConfigValid method returns false when the config is empty.
+    /// This is to test no exception gets thrown with validate during failure config deserialization.
+    /// </summary>
+    [TestMethod]
+    public void TestValidateWithEmptyConfig()
```

Comment:
Shouldn't we add another test case which would have failed with the datasourceName not found exception without your fix?

Reply:
added a test which test the whole deserilization to validation.